### DISCUSSION TITLE
weird question: use Eto with python, no I don't mean in rhino

### DISCUSSION THREAD

**Discussion by retsyo** [picoe/Eto#2571] [Q&A]
As the title says, is there a way to use Eto in python not in Rhino? Thanks
My first try
```python
import clr
import System
clr.AddReference("Eto.WinForms")


import Eto
```
says
```
    import Eto
ModuleNotFoundError: No module named 'Eto'
```

**Comment by cwensley** [comment]
You should be able to.. You'll need to reference both Eto and the platform assembly for it to work.  You'll also need to get all of the dependencies for the platform.  E.g. Eto.WinForms requires both `Microsoft.WindowsAPICodePack.Shell.dll` and `Microsoft.WindowsAPICodePack.dll`.

```python
import clr

clr.AddReference('Eto')
clr.AddReference('Eto.WinForms')

from Eto.Drawing import *
from Eto.Forms import *
from Eto.WinForms import *

class MyForm(Form):
    def __init__(self):
        # Create child controls and initialize form
        self.Content = Label(Text = "Hello IronPython")

app = Application(Platform())
app.Run(MyForm())
```

**Reply by retsyo** [reply]
Thanks

Actually I know very very little about .NET. On my windows 10 64bits, I have got 
- `Eto.WinForms.dll` from `Eto.Platform.Windows.2.8.1.nupkg`
- `Eto.dll` from `Eto.Forms.2.8.1.nupkg`
in https://github.com/picoe/Eto/releases/tag/2.8.1

And JetBrians does not show missing dependencies for `Eto.WinForms.dll` and `Eto.dll`
but the above code still says
```
    from Eto.Drawing import *
ModuleNotFoundError: No module named 'Eto'
```

**Reply by cwensley** [reply]
Again you'll need the additional .dll's as those nuget packages pull in those dependencies. Since you're not using .NET tooling you have to do this manually.  The above I tested using both IronPython 2.7 and 3.4 and it works for me on Windows 11 so something must be amiss, are you using IronPython or are you trying to use some other python?

**Comment by retsyo** [comment]
Thanks, IronPython prompts more detailed information. While pythonnet, which is used by me, does not.
Now I should go to learn something about how to install .NET package correctly.
